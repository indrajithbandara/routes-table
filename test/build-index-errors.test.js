const fs = require('fs');
const path = require('path');
const build = require('../').build;

describe('build-index-errors', () => {
  let structuresDir = path.join(__dirname, 'build-index-errors');
  fs.readdirSync(structuresDir).forEach(structureName => {
    it(structureName, async () => {
      let error;
      let structureDir = path.join(structuresDir, structureName);
      try {
        await build(structureDir, { index:true });
      } catch(e) {
        error = e;
      }

      if (!error) throw new Error('Expected Error');
      expect(error).toMatchSnapshot();
    });
  });
});
