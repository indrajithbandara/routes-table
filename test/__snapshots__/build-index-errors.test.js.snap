// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build-index-errors empty-routes 1`] = `[Error: Expected a \`route.js\`, template, or \`routes/\` directory at test/build-index-errors/empty-routes/routes/file]`;

exports[`build-index-errors no-routes 1`] = `[Error: Expected a \`route.js\`, template, or \`routes/\` directory at test/build-index-errors/no-routes]`;
